resources/assets/js/store/modules/permissions.js

```diff
@@ -0,0 +1,162 @@
+import api from 'plugins/http/api';
+
+/**
+ * Simple interface for interacting with user roles and permissions
+ * 
+ * permissions is an array with the following structure
+ * 
+[
+		{
+			"name": "Create Subsites",
+			"slug": "subsite.create",
+			"roles": [
+				"site.owner"
+			]
+		},
+		...
+ */
+const state = {
+	roles: [],
+	currentRole: '',
+	globalRole: '',
+};
+
+const getters = {
+	getCurrentRole(state, getters) {
+		return state.currentSiteRole;
+	},
+
+	getRoles(state, getters) {
+		return state.roles;
+	},
+
+	/**
+	 * can the user perform the requested action
+	 * if the user is a global admin then let them do anything
+	 * 
+	 * @param {object} state - the vuex state
+	 * @param {string} the permission to check i.e. subsite.edit, subsite.create
+	 * @returns {bool} true or false
+	 */
+	canUser: (state, getters) => (permissionSlug) => {
+
+		let permitted = false;
+		
+		// if the user has the global role of admin then then they can do anything
+		if (state.globalRole === 'admin') {
+			return true;
+		}
+		const matchedRole = state.roles.find(function(value) {
+			if (value.slug == this.permissionSlug) {
+				return true;
+			}
+		}, {permissionSlug});
+			
+		if (matchedRole) {
+			permitted = matchedRole.roles.includes(state.currentRole);
+		}
+		
+		return permitted;   
+	}
+};
+
+const actions = {
+
+	/**
+	 * gets list of roles from the api can calls mutation to store this in the state
+	 * called from TopBar.vue->mounted
+	 * 
+	 * @param {object} state - the vuex state
+	 */
+	loadPermissions({commit, state}) {		
+		api
+			.get('permissions?include=roles')
+			.then((response) => {
+				let permissions = response.data.data;
+				commit('setPermissions', permissions);
+			})
+	},
+
+	/**
+	 * gets list of a user's roles from the api can calls mutation to store this in the state
+	 * called from Editor.vue->created
+	 * 
+	 * @param {object} state - the vuex state
+	 * @param {object}	payload
+	 * @param {int}		payload.site_id  - the id of the current site
+	 * @param {string}	payload.username -  the name of the user
+	 */
+	loadSiteRole({commit, state}, payload) {
+		api
+			.get(`sites/${payload.site_id}?include=users`)
+			.then(({data}) => {
+				const userList = data.data.users;
+				if (userList) {
+					const currentUser = userList.find((element) => element.username === payload.username);
```

Comment:
You should be able to retrieve the global_role for the user from here as well as user.global_role

https://unikent.github.io/astrodocs/api.html#siteuserrole

NOTE the docs need updating to call it user.global_role instead of user.role.

If that isn't the case then the API is wrong and needs fixing.

Reply:
user might not have a role in the site at all, esp. if they are an admin user

Reply:
ah that makes sense :)